### PR TITLE
changing search method

### PR DESCRIPTION
**Pull request by ErfanGerami**

### CODE REVIEW

**Comment by ErfanGerami** [comment]
I used a visited list to have track of visited nodes and the marking methods are not used but they are there because someone else might use them

File: graph.py [hunk 1]
```diff
@@ -1,42 +1,119 @@
 class graph():
     def __init__(self,initial_dic={}):
         self.network=initial_dic
+        self.marks=[]
+        if(len(initial_dic)!=0):
+            self.maximum_node=max(initial_dic)
+        else:
+            self.maximum_node=-1
+        self.nodes=[0 for i in range(self.maximum_node)]
```
Comment (mamadmr) [inline graph.py:9]: intial a constant value for max node for example 400

Comment (mamadmr) [reply graph.py:9]: it makes your code faster ( in the future we may use numpy which use array with constant sizes)

File: graph.py [hunk 2]
```diff
@@ -1,42 +1,119 @@
...
+
     def AddNode(self,node):
+        '''note that nodes must be positive'''
```
Comment (mamadmr) [inline graph.py:12]: are you sure that the node doens't exist?

File: graph.py [hunk 3]
```diff
@@ -1,42 +1,119 @@
...
     def AddEdge(self,node1,node2):
         self.network[node1].add(node2)
         self.network[node2].add(node1)
         
     def DeleteNode(self,node):
         for i in self.network[node]:
-            self.network[i].remove[node]
+            self.network[i].remove(node)
         self.network.pop(node)
+        if(self.maximum_node==node):
```
Comment (mamadmr) [inline graph.py:31]: I think we don't need these codes anymore 
I'm not sure about the purpose of these lines

File: graph.py [hunk 4]
```diff
@@ -1,42 +1,119 @@
...
     def GetConnections(self,node):
         '''returns a set that contains the nodes that the given node is connected to'''
         return self.network[node]
-                
+    
+    #---------------------------------------------
+    def StartMarking(self):
+        '''use to reset or start Marking'''
+        
+        self.marks=[ 0 for i in range(self.maximum_node)]
```
Comment (mamadmr) [inline graph.py:55]: don't forget th change this

File: graph.py [hunk 5]
```diff
@@ -1,42 +1,119 @@
...
+
+    def Mark(self,node,color):
```
Comment (mamadmr) [inline graph.py:57]: for simplify the codes assume that your array is 1-base 
just ignore the first node :)

File: graph.py [hunk 6]
```diff
@@ -1,42 +1,119 @@
...
+        self.marks[node-1]=color
+
+    def GetMark(self,node):
+        return self.marks[node-1]
```
Comment (mamadmr) [inline graph.py:61]: 1-base

File: graph.py [hunk 7]
```diff
@@ -1,42 +1,119 @@
...
+    
+    def FindPath(self,start,dest):
```
Comment (mamadmr) [inline graph.py:63]: this code is very ineffiecient ask me in telegram :)

**Pull request closed without merging**